evil-jumps.el

```diff
@@ -165,10 +165,16 @@ Otherwise the jump commands act only within the current buffer."
           (setf (evil-jumps-struct-idx (evil--jumps-get-current)) idx)
           (run-hooks 'evil-jumps-post-jump-hook))))))
 
+(defun evil--jumps-current-file-name ()
+  "Get the current buffer file name for evil--jumps-push"
+  (or (buffer-file-name)
```

Comment:
This should be
```elisp
(or buffer-file-name
    (when (derived-mode-p 'dired-mode) default-directory))
```
I think. In the cases where `dired-directory` differs from `default-directory` the code will break anyway.

Also, could this be inlined, or do you need it for adding a piece of advice on?